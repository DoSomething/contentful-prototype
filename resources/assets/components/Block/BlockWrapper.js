import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { Link } from 'react-router-dom';
import './block-wrapper.scss';

const BlockTitle = (props) => {
  const title = props.id ? <Link to={`/blocks/${props.id}`}>{props.title}</Link> : props.title;

  return <h4 className="block-wrapper__title">{title}</h4>;
};

BlockTitle.propTypes = {
  title: PropTypes.string,
  id: PropTypes.string,
};

BlockTitle.defaultProps = {
  title: null,
  id: null,
};

const BlockWrapper = props => (
  <div className={classNames('block-wrapper', props.className)}>
    { props.title ? <BlockTitle title={props.title} id={props.id} /> : null }
    {props.children}
  </div>
);

BlockWrapper.propTypes = {
  className: PropTypes.string,
  title: PropTypes.string,
  id: PropTypes.string,
  children: PropTypes.node.isRequired,
};

BlockWrapper.defaultProps = {
  className: null,
  title: null,
  id: null,
};

export default BlockWrapper;
