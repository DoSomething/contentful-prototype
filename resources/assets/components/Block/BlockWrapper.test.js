import React from 'react';
import { mount, shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { MemoryRouter } from 'react-router';
import BlockWrapper from './BlockWrapper';

test('it renders correctly with children', () => {
  const component = shallow(
    <BlockWrapper>
      <div id="tongue-cat" />
    </BlockWrapper>,
  );

  expect(shallowToJson(component)).toMatchSnapshot();
});

test('it renders without children', () => {
  const component = shallow(<BlockWrapper />);

  expect(shallowToJson(component)).toMatchSnapshot();
});

test('it renders correctly with title', () => {
  const component = shallow(
    <BlockWrapper title="Lets Do This">
      <div id="tongue-cat" />
    </BlockWrapper>,
  );

  expect(shallowToJson(component)).toMatchSnapshot();
});

test('it links to a block', () => {
  const wrapper = mount(
    <MemoryRouter>
      <BlockWrapper title="Lets Do This" shareLink="/blocks/123abc">
        <div id="tongue-cat" />
      </BlockWrapper>
    </MemoryRouter>,
  );

  // Grab the link and make sure URL is formed correctly.
  const linkProps = wrapper
    .find('Link')
    .first()
    .props();
  expect(linkProps.to).toEqual('/blocks/123abc');
});
