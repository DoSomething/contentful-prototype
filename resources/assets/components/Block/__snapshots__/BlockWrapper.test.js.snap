// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly with children 1`] = `
<article
  className="block-wrapper"
>
  <div
    id="tongue-cat"
  />
</article>
`;

exports[`it renders correctly with title 1`] = `
<article
  className="block-wrapper"
>
  <BlockTitle
    title="Lets Do This"
  />
  <div
    id="tongue-cat"
  />
</article>
`;

exports[`it renders without children 1`] = `
<article
  className="block-wrapper"
/>
`;
